src/v3/plugins/github/graphView.js

```diff
@@ -0,0 +1,274 @@
+// @flow
+
+import stringify from "json-stable-stringify";
+
+import * as GN from "./nodes";
+import * as GE from "./edges";
+
+import {
+  Graph,
+  type NodeAddressT,
+  Direction,
+  type NeighborsOptions,
+  NodeAddress,
+  EdgeAddress,
+} from "../../core/graph";
+
+export class GraphView {
+  _graph: Graph;
+  _isCheckingInvariants: boolean;
+
+  constructor(graph: Graph) {
+    this._graph = graph;
+    this._isCheckingInvariants = false;
+    this._maybeCheckInvariants();
+  }
+
+  graph(): Graph {
+    this._maybeCheckInvariants();
+    return this._graph;
+  }
+
+  *_nodes<T: GN.StructuredAddress>(prefix: GN.RawAddress): Iterator<T> {
+    for (const n of this._graph.nodes({prefix})) {
+      const structured = GN.fromRaw((n: any));
+      this._maybeCheckInvariants();
+      yield (structured: any);
+    }
+    this._maybeCheckInvariants();
+  }
+
+  *_neighbors<T: GN.StructuredAddress>(
+    node: GN.StructuredAddress,
+    options: NeighborsOptions
+  ): Iterator<T> {
+    if (!NodeAddress.hasPrefix(options.nodePrefix, GN._Prefix.base)) {
+      throw new Error(`_neighbors must filter to Github nodes`);
+    }
+    const rawNode = GN.toRaw(node);
+    for (const neighbor of this._graph.neighbors(rawNode, options)) {
+      this._maybeCheckInvariants();
+      yield (GN.fromRaw((neighbor.node: any)): any);
+    }
+    this._maybeCheckInvariants();
+  }
+
+  _children<T: GN.StructuredAddress>(
+    node: GN.StructuredAddress,
+    nodePrefix: GN.RawAddress
+  ): Iterator<T> {
+    const options = {
+      nodePrefix,
+      edgePrefix: GE._Prefix.hasParent,
+      direction: Direction.IN,
+    };
+    return this._neighbors(node, options);
+  }
+
+  repos(): Iterator<GN.RepoAddress> {
+    return this._nodes(GN._Prefix.repo);
+  }
+
+  issues(repo: GN.RepoAddress): Iterator<GN.IssueAddress> {
+    return this._children(repo, GN._Prefix.issue);
+  }
+
+  pulls(repo: GN.RepoAddress): Iterator<GN.PullAddress> {
+    return this._children(repo, GN._Prefix.pull);
+  }
+
+  comments(commentable: GN.CommentableAddress): Iterator<GN.CommentAddress> {
+    return this._children(commentable, GN._Prefix.comment);
+  }
+
+  reviews(pull: GN.PullAddress): Iterator<GN.ReviewAddress> {
+    return this._children(pull, GN._Prefix.review);
+  }
+
+  // TODO figure out how to overload this fn signature
```

Comment:
Could you `TODO(@someone)` here, please?

Reply:
You get a TODO! :)